spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -204,6 +205,41 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   }
 
   /**
+   * set trainable. If trainable is false, its parameters(weight/bias, if exists)
+   * are not changed in training process
+   * @param trainable whether this layer is trainable
+   */
+  def setTrainable(trainable: Boolean): this.type = {
+    if (!trainable) {
+      scaleW = 0
+      scaleB = 0
+    }
+    this
+  }
+
+  private var _stopGradient: Boolean = false
+
+  /**
+   * whether stop propagating gradInput back
+   * @return
+   */
+  def isStopGradient(): Boolean = {
+    _stopGradient
+  }
+
+  /**
+   * set stop gradient. If isStopGradient is true, its input gradient
+   * is not computed. And it will not contributed to the input gradient computation of
+   * layers that depend on this layer.
+   * @param isStopGradient default is true.
+   * @return
+   */
+  def setStopGradient(isStopGradient: Boolean = true): this.type = {
```

Comment:
after user build a graph, he/she change the layer/trainable or stopGradient. Is this right?